README.md

```diff
@@ -1,207 +1,249 @@
+| data_type  | enum      | **REQUIRED.** The data type of values in the n-dimensional array. For model inputs, this should be the data type of the processed input supplied to the model inference function, not the data type of the source bands. Use one of the [common metadata data types](https://github.com/stac-extensions/raster?tab=readme-ov-file#data-types). |     |
+
+Note: It is common in the machine learning, computer vision, and remote sensing communities to refer to rasters that are inputs to a model as arrays or tensors. Array Objects are distinct from the JSON array type used to represent lists of values.
+
+### Model Output Object
+
+| Field Name               | Type                                          | Description                                                                                                                                                                                            |
+| ------------------------ | --------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| task                     | [Task Enum](#task-enum)                       | **REQUIRED.** Specifies the Machine Learning task for which the output can be used for.                                                                                                                |
+| result_array             | [[Result Array Object](#result-array-object)] | The list of output arrays/tensors from the model.                                                                                                                                                      |
+| classification:classes   | [[Class Object](#class-object)]               | A list of class objects adhering to the [Classification extension](https://github.com/stac-extensions/classification).                                                                                 |
+| post_processing_function | string                                        | A url to the postprocessing function where normalization, rescaling, and other operations take place.. Or, instead, the function code path, for example: `my_package.my_module.my_processing_function` |
+
+
+While only `task` is a required field, all fields are recommended for supervised tasks that produce a fixed shape tensor and have output classes.
+`image-captioning`, `multi-modal`, and `generative` tasks may not return fixed shape tensors or classes.
 
 #### Task Enum
 
-It is recommended to define `task` with one of the following values:
+It is recommended to define `task` with one of the following values for each Model Output Object:
 - `regression`
 - `classification`
 - `object detection`
-- `segmentation` (generic)
```

Comment:
There is often confusion between `classification` and `segmentation` in terms of the point of view under which the predictions are considered (ie: patch-wise vs pixel-wise). 
More specifically, if someone has an algorithm that does pixel-wise "classification", the resulting prediction could be considered "semantic segmentation" when looking at the patch as a whole. 
We should replace `classification` by an explicit `scene classification` to avoid this ambiguity.
A description of each type of task wouldn't hurt as well to remove any possible confusion.

Reply:
https://github.com/rbavery/dlm-extension/pull/2